utils/remote-ext-tests/replay-block/src/main.rs

```diff
@@ -0,0 +1,221 @@
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+//! Replaying a block against the live chain state
+
+use clap::Parser;
+use codec::Encode;
+use runtime_primitives::Block;
+#[cfg(feature = "always-wasm")]
+use sc_executor::sp_wasm_interface::ExtendedHostFunctions;
+use sc_tracing::logging::LoggerBuilder;
+#[cfg(all(not(feature = "always-wasm"), feature = "gear-native"))]
+use service::GearExecutorDispatch;
+#[cfg(all(not(feature = "always-wasm"), feature = "vara-native"))]
+use service::VaraExecutorDispatch;
+use sp_runtime::{
+    generic::SignedBlock,
+    traits::{Block as BlockT, Header as HeaderT, One},
+};
+use sp_state_machine::ExecutionStrategy;
+use std::fmt::Debug;
+use substrate_rpc_client::{ws_client, ChainApi};
+use util::*;
+
+pub(crate) const LOG_TARGET: &str = "remote-ext::cli";
+
+mod parse;
+mod util;
+
+const VARA_SS58_PREFIX: u8 = 137;
+const GEAR_SS58_PREFIX: u8 = 42;
+
+pub(crate) type HashFor<B> = <B as BlockT>::Hash;
+pub(crate) type NumberFor<B> = <<B as BlockT>::Header as HeaderT>::Number;
+
+#[derive(Clone, Debug)]
+pub(crate) enum BlockHashOrNumber<B: BlockT> {
+    Hash(HashFor<B>),
+    Number(NumberFor<B>),
+}
+
+#[derive(Clone, Debug, Parser)]
+struct Opt {
+    /// The RPC url.
+    #[arg(
+		short,
+		long,
+		value_parser = parse::url,
+		default_value = "wss://archive-rpc.vara-network.io:443"
+	)]
+    uri: String,
+
+    /// The block hash or number we want to replay. If omitted, the latest finalized block is used.
+    /// The blockchain state at previous block with respect to this parameter will be scraped.
+    #[arg(
+		short,
+		long,
+		value_parser = parse::block,
+	)]
+    block: Option<BlockHashOrNumber<Block>>,
+
+    /// Pallet(s) to scrape. Comma-separated multiple items are also accepted.
+    /// If empty, entire chain state will be scraped.
+    #[arg(short, long, num_args = 1..)]
+    pallet: Vec<String>,
+
+    /// Fetch the child-keys as well.
+    ///
+    /// Default is `false`, if specific `--pallets` are specified, `true` otherwise. In other
+    /// words, if you scrape the whole state the child tree data is included out of the box.
+    /// Otherwise, it must be enabled explicitly using this flag.
+    #[arg(long)]
+    child_tree: bool,
+
+    /// Sets a custom logging filter. Syntax is `<target>=<level>`, e.g. -lsync=debug.
+    ///
+    /// Log levels (least to most verbose) are error, warn, info, debug, and trace.
+    /// By default, all targets log `info`. The global log level can be set with `-l<level>`.
+    #[arg(short = 'l', long, value_name = "LOG_PATTERN", num_args = 1..)]
```

Comment:
```suggestion
    #[arg(short = 'l', long, value_name = "NODE_LOG", num_args = 0..)]
```

Reply:
Same as RUST_LOG

Reply:
Why `num_args = 0..`? Do we want to allow a dangling `-l` flag with no target?

Reply:
> Why `num_args = 0..`? Do we want to allow a dangling `-l` flag with no target?

for example if we will throw some env var like `-l LOG_ENV` so why not